specs/typography/typography.md

```diff
@@ -0,0 +1,143 @@
+- CSS variables to apply the type ramp
+
+### Risks and Challenges
+
+Further typography-related specs build on the underlying principles and resources defined here. It will be important to provide meaningful ways to allow for semantic styles and consistent application of appearances in balance with flexibility and maintenance cost.
+
+### Prior Art / Examples
+
+- [FAST-DNA (React)](https://explore.fast.design/components/typography)
+- [Material Design](https://material.io/design/typography/the-type-system.html)
+- [Bootstrap](https://getbootstrap.com/docs/4.4/content/typography)
+- [Ant Design](https://ant.design/components/typography)
+- [Mineral UI](https://mineral-ui.netlify.com/typography)
+
+---
+
+## Design
+
+The default type ramp uses a selection of values from [Microsoft Fluent Design](https://www.microsoft.com/design/fluent).
+
+![](./images/type-ramp.png)
+
+Note the "Ramp ID", which is used to select the desired size. This ramp is centered around the base type size to associate the relative sizes between content. For instance, your main body text uses the `base` size. You may want a caption to be smaller than that, so you use `-1`. You want your [headings](./headings.md) to be larger, so you use one or more of the `+` sizes.
+
+This is more declarative than using an index position to reference the sizes, like `7` if this ramp is numbered from the top or `3` if numbered from the bottom. It also allows the sizes that you use to be related to each other rather than at other arbitrary points on the ramp. For example, three levels of headings above body text are `16px`, `20px`, and `28px` instead of jumping all the way to `60px` at the top and skipping the middle. You can add additional sizes at the top and bottom if necessary.
+
+*Note that these are just examples and nothing about any particular usage of a type size is prescribed by the underlying ramp. See further typography specs for other systematic ways to apply the sizes.*
+
+Type style variables can be added at any point in the DOM. Generally they would be created globally for consistent application throughout your app or site, or they could be overridden or added for a single section of a page.
+
+### API
+
+Pseudo interfaces: (*Could be functions or decorators or otherwise*)
+
+- `typeRamp()`: A default type ramp as illustrated above
+- `typeRamp({'12px', '14px', '24px', '36px'}, 1)`: A type ramp with fixed type size values and default line height, index 1 is base
+- `typeRamp({{'12px', '16px'}, {'14px', '20px'}, {'24px', '28px'}, {'36px', '44px'}}, 1)`: A type ramp with fixed type size values and fixed individual line heights
+- `typeRamp(16, 1.25, 3, 5)`: A type ramp based at `16px` on a major third [modular scale](https://type-scale.com/) of `1.25` with `3` sizes below base and `5` sizes above.
+
+### Anatomy and Appearance
+
+The type ramp is presented as css variables that extend plus and minus from the base.
+
+- `fast-type-ramp-base-size` = `"14px"`
```

Comment:
Where do these properties come from? Are they defined by the design-system-provider component?

Reply:
Good question that made me realize I was essentially thinking of this as a fragment of the design system specifically for type sizes. I suspect there are plenty of reasons that we'd want or need to keep at least the minimal values used by the default FAST components in the main design system provider though.

I'll consider some simplification here, but also I'm now reconsidering the significance of the underlying ramp outside of the ways it's used or mapped to components. I'm going to get back to a test page real quick and review the usage in the main components.